### PR TITLE
Patch for COCOReader - new IDs of labels

### PR DESCRIPTION
**Pull request by wojciechpr**
Signed-off-by: wojciechp <wojciechp@nvidia.com>

### CODE REVIEW

**Comment by cliffwoolley** [comment]
What does this do, exactly?

**Comment by wojciechpr** [comment]
I've been using COCOReader from pull request 110 and that's my diff for a file coco_reader_op.h. As git doesn't support pr for another pr I've added just my diff. I've discussed it with Przemek Tredak.

Although COCO has 80 labels (+background) they don't have IDs from 1 to 80. In fact it is some subset of integers from 1 to 90. As SSD expects numbers from 1 to 80 we have to map them on this numbers.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by cliffwoolley** [comment]
You can just add this commit on top of the branch from #110 and GitHub should handle it properly...

**Comment by wojciechpr** [comment]
Oh okay, sorry for that

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by JanuszL** [comment]
So please do as @cliffwoolley said and then close this PR.

**Comment by JanuszL** [comment]
@wojciechpr - one more comment as I understand this is very SSD specific patch, so other networks that uses COCO reader may not work with that? If so, how about providing some vector, as an argument, to COCO reader with deleted categories?

**Comment by wojciechpr** [comment]
As far as I know it's common to do it also in other models. I think that specifying in COCOReader constructor if we want it to map labels' IDs would be good solution.

________________________________

File: dali/pipeline/operators/reader/coco_reader.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+80,92d79
+<       // Change categories IDs to be in range [1, 80]
+<       std::vector<int> deleted_categories{ 12, 26, 29, 30, 45, 66, 68, 69, 71, 83, 91 };
+<       std::map<int, int> new_categories_ids;
+<       int current_id = 1;
+<       int vector_id = 0;
+<       for (int i = 1; i <= 90; i++) {
```
Comment (cliffwoolley) [inline dali/pipeline/operators/reader/coco_reader.patch]: Why 90 instead of 91?

**Comment by wojciechpr** [comment]
90 is the ID of last class present in COCO. 91 in vector is in fact guardian so deleted_categories[vector_id] is always valid.

________________________________

**Comment by wojciechpr** [comment]
I'm afraid I can't commit to PR 110 as I don't have push access for branch coco_reader

**Comment by wojciechpr** [comment]
I get permission denied (publickey) error

**Comment by wojciechpr** [comment]
Oh, I've probably managed to create pr to that branch, so if everything is okay I will close this pr.

**Comment by JanuszL** [comment]
#110 update. Closing that one.

**Pull request closed without merging**